reference/Interpreter.cpp

```diff
@@ -0,0 +1,99 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "Interpreter.h"
+
+#include "Ops.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/Support/Errc.h"
+#include "mlir/IR/Block.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Support/DebugStringHelper.h"
+#include "dialect/StablehloOps.h"
+
+namespace mlir {
+namespace stablehlo {
+
+namespace {
+
+template <typename... Ts>
+inline llvm::Error invalidArgument(char const *Fmt, const Ts &...Vals) {
+  return createStringError(llvm::errc::invalid_argument, Fmt, Vals...);
+}
+
+}  // namespace
+
+llvm::Expected<SmallVector<Tensor>> eval(func::FuncOp func,
+                                         ArrayRef<Tensor> args) {
+  if (func->getNumRegions() != 1) {
+    return invalidArgument("Expected one region in func %s",
```

Comment:
Will StableHLO/interpreter follow MLIR/LLVM's error conventions?

Reply:
Will be addressed via 
https://github.com/openxla/stablehlo/issues/15